Implemented IpAddress type #59

dejobratic opened the pull request:
- Created IpAddress value object with string as underlying type.
- IpAddress is validated and created based on the .NET's System.Net.IPAddress class.

Commit:
created ip address type;

Commit:
added converter;

Commit:
resolved merge conflict;

rochar commented on tests/Nox.Types.Tests/Types/IpAddress/IpAddressTests.cs:
```diff
@@ -3,9 +3,88 @@ namespace Nox.Types.Tests.Types;
+    [InlineData("172.16.0.0")]
+    [InlineData("192.0.2.1")]
+    [InlineData("255.255.255.255")]
+    public void Nox_IpAddress_Constructor_WithIPv4Value_ReturnsSameValue(string ipV4Value)
```
Please remove the Nox prefix (we are starting doing this since is redundant)

Also see our guidelines that will be approved https://regusit.atlassian.net/wiki/spaces/NOX/pages/3153461266/Coding+Conventions+Styles+and+Guidelines and see this one: https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#best-practices its simple and defined in .Net ... 

```suggestion
    public void IpAddress_Constructor_WithIPv4Value_ReturnsSameValue(string ipV4Value)
```

dejobratic replied:
Thanks for the suggestions. Will apply them now.

rochar reviewed (changes requested):
looks good, just a small comment

thks

Commit:
applied changes according to coding conventions and standards;

jan-schutte committed:
Merge remote-tracking branch 'origin/main' into feature/ip-address-type

# Conflicts:
#	tests/Nox.Types.Tests/EntityFrameworkTests/NoxTypesEntityFrameworkTests.cs

jan-schutte committed:
- merged main into current
- fixed time format in ToString_WithCultureParameter_ReturnsFormattedString test

jan-schutte committed:
- reverted time format in ToString_WithCultureParameter_ReturnsFormattedString
